### ISSUE TITLE
Circles don't look sharp

### ISSUE THREAD

**Issue by abique** [slint-ui/slint#4831]
It feels to me that the painting could look better:

![Screenshot from 2024-03-12 22-23-06](https://github.com/slint-ui/slint/assets/31200/1b1023e5-bcb5-4429-ae30-7c7cb0e06ebc)

Using skia.

**Comment by Enyium** [comment]
That's right. Rounded `Rectangle` corners aren't antialiased well enough.

[SlintPad example](https://slintpad.com/?snippet=export+component+Demo+%7B%0A++++Rectangle+%7B%0A++++++++property+%3Clength%3E+size%3A+min%28parent.width%2C+parent.height%29+*+90%25%3B%0A++++++++width%3A+self.size%3B%0A++++++++height%3A+self.size%3B%0A++++++++border-radius%3A+self.size+%2F+2%3B%0A++++++++background%3A+black%3B%0A++++%7D%0A%7D%0A&style=native) (change the circle's size via the handles in the corners).

To compare this with an SVG `<circle>`, visit <https://developer.mozilla.org/en-US/docs/Web/SVG/Element/circle>, click "Play" on the circle preview and paste the following JavaScript code to make the circle change its size together with your mouse position:

```js
const svg = document.getElementsByTagName('svg')[0];
const circle = document.getElementsByTagName('circle')[0];

svg.addEventListener('mousemove', function(event) {
    const rect = svg.getBoundingClientRect();
    const viewBox = svg.viewBox.baseVal;
    
    const scaleX = viewBox.width / rect.width;
    const scaleY = viewBox.height / rect.height;

    const cx = parseFloat(circle.getAttribute('cx'));
    const cy = parseFloat(circle.getAttribute('cy'));
    
    const mouseX = (event.clientX - rect.left) * scaleX;
    const mouseY = (event.clientY - rect.top) * scaleY;

    const dx = mouseX - cx;
    const dy = mouseY - cy;

    const newRadius = Math.sqrt(dx * dx + dy * dy);
    circle.setAttribute('r', newRadius);
});
```

**Comment by tronical** [comment]
I'm having difficulties reproducing this with skia (as @abique originally tagged it as). Running @Enyium 's example with Skia on my desktop (note that Slintpad doesn't use Skia) I get this result:
<img width="15" alt="Screenshot 2024-09-12 at 11 22 45" src="https://github.com/user-attachments/assets/7b0c4530-3cec-4f94-9a8b-4182364380ab">

That looks alright, doesn't it?

**Comment by tronical** [comment]
In contrast, this is the result with FemtoVG:

<img width="15" alt="Screenshot 2024-09-12 at 11 24 25" src="https://github.com/user-attachments/assets/96ee0e18-9ac8-48ea-bda5-09ec0e575ad4">


This isn't quite as nice as Skia, the quality of the anti-aliasing is not as good.

**Comment by Enyium** [comment]
Yeah, I'm sorry, Skia renders the circle very well in a native Windows 10 window. I can't make out any quality differences to the SVG circle.

Even the software renderer seems on par with Skia in this regard.

FemtoVG circles have lower quality:

![femtovg-circle](https://github.com/user-attachments/assets/8bd31eef-f42c-4cec-80b4-a3620428858f)

**Comment by abique** [comment]
Try rendering to bitmap and compare with what cairo can do.

**Comment by abique** [comment]
I think skia can do a 4x oversampling right?

**Comment by expenses** [comment]
Possibly related to #9310?
